internal/distro/fedora30/distro_test.go

```diff
@@ -137,3 +149,313 @@ func TestImageType_BuildPackages(t *testing.T) {
 		}
 	}
 }
+
+func TestImageType_Name(t *testing.T) {
+	distro := fedora30.New()
+	imgMap := []struct {
+		arch     string
+		imgNames []string
+	}{
+		{
+			arch: "x86_64",
+			imgNames: []string{
+				"ami",
+				"ext4-filesystem",
+				"partitioned-disk",
+				"qcow2",
+				"openstack",
+				"tar",
+				"vhd",
+				"vmdk",
+			},
+		},
+		{
+			arch: "aarch64",
+			imgNames: []string{
+				"ami",
+				"ext4-filesystem",
+				"partitioned-disk",
+				"qcow2",
+				"openstack",
+				"tar",
+			},
+		},
+	}
+	for _, mapping := range imgMap {
+		arch, err := distro.GetArch(mapping.arch)
+		assert.Nil(t, err)
+		if err == nil {
+			for _, imgName := range mapping.imgNames {
+				imgType, err := arch.GetImageType(imgName)
+				assert.Nil(t, err)
+				if err == nil {
+					assert.Equalf(t, imgName, imgType.Name(), "arch: %s", imgType.Name())
```

Comment:
Shouldn't the last argument be `arch`, or was it wrong before ?

Reply:
Yes, thanks for noticing this.